### PR TITLE
billing control nos

### PR DESCRIPTION
**Pull request by stephenwaite**
[forum thread](https://community.open-emr.org/t/report-eligibility-270-inquiry-batch/1782/52?u=stephenwaite)

### CODE REVIEW

**Comment by stephenwaite** [comment]
also fixed a query for payment select that wasn't respecting de-activated list options
plus a php8 fix and a typo

File: interface/modules/zend_modules/public/js/installer/action.js [hunk 1]
```diff
@@ -218,7 +218,7 @@ function DeleteHooks(hooksID,mod_id,msg){
 }
 
 /**
-* Save Settings Tab Contants
+* Save Settings Tab Contents
```
Comment (bradymiller) [inline interface/modules/zend_modules/public/js/installer/action.js:221]: 2DV

File: src/Billing/BillingProcessor/BillingClaimBatch.php [hunk 2]
```diff
@@ -228,7 +234,7 @@ public function append_claim(&$segs)
 
             if ($elems[0] == 'BHT') {
                 // needle is set in OpenEMR\Billing\X125010837P
-                $this->bat_content .= substr_replace($seg, '*' . $this->bat_icn . $bat_st_02 . '*', strpos($seg, '*0123*'), 6);
+                $this->bat_content .= substr_replace($seg, '*' . $this->bat_gs06 + 1 . '*', strpos($seg, '*0123*'), 6);
```
Comment (bradymiller) [inline src/Billing/BillingProcessor/BillingClaimBatch.php:237]: will this keep the padding that you need? (ie. does it stay there after it is converted to an into to be able to add 1 to it)

Comment (bradymiller) [reply src/Billing/BillingProcessor/BillingClaimBatch.php:237]: oh wait, this is a replace, so forget my comment above. just move along :)

File: src/Billing/BillingProcessor/BillingClaimBatch.php [hunk 3]
```diff
@@ -210,7 +215,8 @@ public function append_claim(&$segs)
             if ($elems[0] == 'GS') {
                 if ($this->bat_gscount == 0) {
                     ++$this->bat_gscount;
-                    $this->bat_content .= "GS*HC*" . $elems[2] . "*" . $elems[3] . "*$this->bat_yyyymmdd*$this->bat_hhmm*1*X*" . $elems[8] . "~";
+                    $this->bat_gs06 = $this->bat_icn + 1;
```
Comment (bradymiller) [inline src/Billing/BillingProcessor/BillingClaimBatch.php:218]: ok, here is where I wonder if the padding will be kept

Comment (stephenwaite) [reply src/Billing/BillingProcessor/BillingClaimBatch.php:218]: you're right, the padding is lost but only the ISA needs it

Comment (stephenwaite) [reply src/Billing/BillingProcessor/BillingClaimBatch.php:218]: I'll make a better comment in the constructor and then note some of the other places where we're using it to increment the other segment counters

**Review by bradymiller — APPROVED** [review]

**Comment by stephenwaite** [comment]
testing well going to bring in, going to look into adding tests for billing

**Pull request merged**